scripts/pre_commit_linter.py

```diff
@@ -881,8 +883,91 @@ def _check_html_directive_name(all_files):
     return summary_messages
 
 
+def _check_directive_scope(all_files):
+    """This function checks that all directives have an explicit
+    scope: {} and it should not be scope: true.
+    """
+    print 'Starting directive scope check'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js')]
+    failed = False
+    directive_has_isolated_scope = False
+    summary_messages = []
+    parser = pyjsparser.PyJsParser()
+    for filename in files_to_check:
+        with open(filename) as f:
+            content = f.read()
+        parsed_nodes = parser.parse(content)['body']
+        for parsed_node in parsed_nodes:
+            if parsed_node['type'] != 'ExpressionStatement':
+                continue
+            expression = parsed_node['expression']
+            if not (expression['type'] == 'CallExpression' and (
+                    expression['callee']['type'] == (
+                        'MemberExpression')) and (
+                            expression['callee']['property']['name'] == (
+                                'directive'))):
+                continue
+            arguments = expression['arguments']
+            directive_name = str(arguments[0]['value']) if (
+                arguments[0]['type'] == 'Literal') else (
+                    "No such directive exists.")
```

Comment:
Use single quotes in Python.

It seems weird for directive_name to be 'No such directive exists." In what cases does this happen and should we be raising an error or throwing an exception instead?

Also, consider dividing up all this logic (for the whole function) into short "blocks" of code and adding a short comment above each block to explain what the block is doing. It's a bit hard to follow.

Reply:
I ran the check again for the complete codebase. I think we do not require the "No such directive exists" check since we are already checking for a directive. Therefore that has been removed. Also, other changes have been made too. Thanks!